Fix memory and slots test

mvdbeek opened the pull request:
Guess I need to start using drafts 😆 

## How to test the changes?
(Select all options that apply)
- [ ] I've included appropriate [automated tests](https://docs.galaxyproject.org/en/latest/dev/writing_tests.html).
- [ ] This is a refactoring of components with existing test coverage.
- [ ] Instructions for manual testing are as follows:
  1. [add testing steps and prerequisites here if you didn't write automated tests covering all your changes]

## License
- [x] I agree to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT).

mvdbeek committed:
Fix memory and slots test

nsoranzo commented on test/functional/tools/galaxy_slots_and_memory.xml:
```diff
@@ -1,6 +1,6 @@
 <tool id="galaxy_slots_and_memory" name="Galaxy Slots and Memory" version="1.0.0" profile="22.05">
     <command><![CDATA[
-echo "\$GALAXY_SLOTS\n\$GALAXY_MEMORY_MB_PER_SLOT\n\$GALAXY_MEMORY_MB" > '$output'
+echo "\$GALAXY_SLOTS\n$GALAXY_MEMORY_MB_PER_SLOT\n$GALAXY_MEMORY_MB" > '$output'
```
```suggestion
echo "$GALAXY_SLOTS\n$GALAXY_MEMORY_MB_PER_SLOT\n$GALAXY_MEMORY_MB" > '$output'
```

I suppose?

mvdbeek replied:
No, that would make cheetah think there's a variable called `GALAXY_SLOTS\n$GALAXY_MEMORY_MB_PER_SLOT\n$GALAXY_MEMORY_MB`

nsoranzo replied:
Ah, sorry, then I think the correct fix is:

```suggestion
printf "%s\n%s\n%s\n" \$GALAXY_SLOTS \$GALAXY_MEMORY_MB_PER_SLOT \$GALAXY_MEMORY_MB > '$output'
```

nsoranzo committed:
Use ``printf``

`echo` doesn't interpret backslash escapes by default, and its `-e` option is not POSIX.

nsoranzo committed:
Add test
